### PR TITLE
Add strain rate particle property

### PR DESCRIPTION
**Pull request by KiralyAgi**
I added a new plugin to particle/property/strainrate, to save the local strain rate on the particles.

### CODE REVIEW

File: include/aspect/particle/property/strainrate.h [hunk 1]
```diff
@@ -0,0 +1,97 @@
+/*
+ Copyright (C) 2015 - 2021 by the authors of the ASPECT code.
```
Comment (MFraters) [inline include/aspect/particle/property/strainrate.h:2]: Please update

File: include/aspect/particle/property/strainrate.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Property
+    {
+      /**
+       * A class that sets particle properties to the current velocity.
```
Comment (MFraters) [inline include/aspect/particle/property/strainrate.h:34]: strain-rate instead of velocity?

File: source/particle/property/strainrate.cc [hunk 3]
```diff
@@ -0,0 +1,105 @@
+/*
+  Copyright (C) 2015 - 2021 by the authors of the ASPECT code.
```
Comment (MFraters) [inline source/particle/property/strainrate.cc:2]: please update

File: source/particle/property/strainrate.cc [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                                                      const Vector<double> &/*solution*/,
+                                                      const std::vector<Tensor<1,dim> > &gradients,
+                                                      typename ParticleHandler<dim>::particle_iterator &particle) const
+      {
+        auto &data = particle->get_properties();
+        // Velocity gradients
+        Tensor<2,dim> grad_u;
+        for (unsigned int d=0; d<dim; ++d)
+          grad_u[d] = gradients[d];
+        //std::cout<<"grad_u: "<<grad_u<<std::endl;
```
Comment (MFraters) [inline source/particle/property/strainrate.cc:54]: please remove this line.

File: source/particle/property/strainrate.cc [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        // Calculate strain rate from velocity gradients
+        const SymmetricTensor<2,dim> strain_rate = symmetrize (grad_u);
+        //std::cout<<"strainrate: "<<strain_rate<<std::endl;
```
Comment (MFraters) [inline source/particle/property/strainrate.cc:57]: remove this line

File: source/particle/property/strainrate.cc [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Property
+    {
+      ASPECT_REGISTER_PARTICLE_PROPERTY(Strainrate,
+                                        "strainrate",
+                                        "Implementation of a plugin in which the particle "
+                                        "property is defined as the recent strainrate at "
```
Comment (MFraters) [inline source/particle/property/strainrate.cc:101]: strain rate is two words. I think we usually write `StrainRate` for the class, `strain_rate` for the file name  and `strain rate` for the string name. See https://github.com/geodynamics/aspect/blob/main/source/postprocess/visualization/strain_rate.cc for an example.

**Review by MFraters — CHANGES_REQUESTED** [review]
A few small changes requested. Could you also indent the code and add a test?

**Comment by KiralyAgi** [comment]
I renamed the files to strain_rate, the class to StrainRate, and the string to strain rate. Added a test.

File: include/aspect/particle/property/strain_rate.h [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Property
+    {
+      /**
+       * A class that sets particle properties to the current strain rate.
```
Comment (MFraters) [inline include/aspect/particle/property/strain_rate.h:34]: Please update this to strain rate from the previous time step

**Review by MFraters — APPROVED** [review]
One small comment, otherwise good to go.

**Comment by gassmoeller** [comment]
Apart from Menno's comment, you will also have to update the test results with the reference results. Let me know if you need help with it (also see here https://aspect-documentation.readthedocs.io/en/latest/user/extending/testing/writing-tests.html).

**Comment by gassmoeller** [comment]
Thank you! Ready to go.

**Pull request merged**